src/components/Messaging/Messaging.js

```diff
@@ -140,17 +149,24 @@ const Messaging = () => {
       <Grid container className={container}>
         <Grid item className={inbox} xs={5} md={4}>
           <Inbox
-            threads={messages}
-            selectedIndex={selectedIndex}
+            threads={threads}
+            selected={messageRecipient}
             handleListItemClick={handleListItemClick}
           />
         </Grid>
         <Grid item className={body} xs={7} md={8}>
-          {messages.length ? (
+          {/* {console.log(
+            'selectedIndex currentThread --- ',
+            selectedIndex,
+            currentThread
+          )} */}
+          {threads.length ? (
             <MessageBody
-              messages={messages[selectedIndex].messages}
-              messageRecipient={messages[selectedIndex].userName}
-              avatar={messages[selectedIndex].avatar}
+              messages={threads.length && threads[selectedIndex].messages}
+              messageRecipient={
+                threads.length && threads[selectedIndex].userName
+              }
+              avatar={threads.length && threads[selectedIndex].avatar}
```

Comment:
I don't think the check for `threads.length` is needed since you already do this check on line 163.